Intermittent Fortran CI Issue.

We are currently seeing sporadic CI failures on jobs that have Fortran enabled. See [here](https://github.com/AMReX-Codes/amrex-tutorials/runs/3510928843) for an example. 

I believe this is related to a dependency problem when running make with multiple processes, i.e. `-j 2`.

Thanks for documenting!

Quickly copying the full output before it gets purged:

```
[ 72%] Building Fortran object EB/CNS/CMakeFiles/CNS_Pulse.dir/Source/fortran/bc_fill_nd.F90.o
cd /home/runner/work/amrex-tutorials/amrex-tutorials/build/EB/CNS && /usr/bin/gfortran  -I/home/runner/work/amrex-tutorials/amrex-tutorials/EB/CNS/Source -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/EB/CNS/_mod_files -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-build/mod_files -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/Base -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/Base/Parser -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/Boundary -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/AmrCore -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/Amr -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/EB -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/LinearSolvers/MLMG -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/LinearSolvers/Projections -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/F_Interfaces/Base -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/F_Interfaces/AmrCore -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/F_Interfaces/Octree -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/F_Interfaces/LinearSolvers -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/F_Interfaces/Particle -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-src/Src/Particle -I/home/runner/work/amrex-tutorials/amrex-tutorials/build/_deps/amrex-build -g -J_mod_files -pthread -O0 -ggdb -fcheck=bounds -fbacktrace -Wuninitialized -Wunused -finit-real=snan -finit-integer=2147483647 -fimplicit-none -c /home/runner/work/amrex-tutorials/amrex-tutorials/EB/CNS/Source/fortran/bc_fill_nd.F90 -o CMakeFiles/CNS_Pulse.dir/Source/fortran/bc_fill_nd.F90.o
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/probdata_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/probdata_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/diffusion_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/diffusion_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/eb_diffusion_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/eb_diffusion_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/cns_eb_diff_wall_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/cns_eb_diff_wall_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/diff_coef_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/diff_coef_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/cns_derive.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/cns_derive.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/cns_eb_hyp_wall_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/cns_eb_hyp_wall_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/slope_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/slope_module.mod.stamp GNU
/usr/local/bin/cmake -E cmake_copy_f90_mod EB/CNS/_mod_files/cns_nd_module.mod EB/CNS/CMakeFiles/CNS_Pulse.dir/cns_nd_module.mod.stamp GNU
Error copying Fortran module "EB/CNS/_mod_files/cns_nd_module.mod".  Tried "EB/CNS/_mod_files/CNS_ND_MODULE.mod" and "EB/CNS/_mod_files/cns_nd_module.mod".
make[2]: *** [EB/CNS/CMakeFiles/CNS_Pulse.dir/cns_nd_module.mod.stamp] Error 1
EB/CNS/CMakeFiles/CNS_Pulse.dir/depend.make:106: recipe for target 'EB/CNS/CMakeFiles/CNS_Pulse.dir/cns_nd_module.mod.stamp' failed
make[2]: *** Waiting for unfinished jobs....
```

[raw.txt](https://github.com/AMReX-Codes/amrex-tutorials/files/7124717/raw.txt)

--- a/cmake/SetupTutorials.cmake
+++ b/cmake/SetupTutorials.cmake
@@ -42,11 +42,11 @@ function (setup_tutorial _srcs  _inputs)
    if (_HAS_FORTRAN_MODULES)
       target_include_directories(${_exe_name}
          PRIVATE
-         ${CMAKE_CURRENT_BINARY_DIR}/${EXENAME}_mod_files)
+         ${CMAKE_CURRENT_BINARY_DIR}/${_exe_name}_mod_files)
       set_target_properties( ${_exe_name}
          PROPERTIES
          Fortran_MODULE_DIRECTORY
-         ${CMAKE_CURRENT_BINARY_DIR}/${EXENAME}_mod_files )
+         ${CMAKE_CURRENT_BINARY_DIR}/${_exe_name}_mod_files )
    endif ()
 
    target_link_libraries( ${_exe_name} AMReX::amrex )

